### PR TITLE
Source Code Updates

### PR DESCRIPTION
**Pull request by lkadian**
#47  

Made updates based on design meeting:
   * Added booking_type attribute to Availability table in DB
   * Added booking_type enum and appropriate methods to ScheduleRequest
   * Updated get_availabilities() to also filter for booking_type in it's query.
   * Added free_availability() method
   * Updated validate_availability_and_reserve()

### CODE REVIEW

File: uber_sante/models/scheduler.py [hunk 1]
```diff
@@ -4,48 +4,65 @@
...
 class RequestEnum(Enum):
-
     DAILY_REQUEST = 1
     MONTHLY_REQUEST = 2
 
 
+class BookingType(Enum):
+    ANNUAL = "ANNUAL"
+    WALKIN = "WALKIN"
+    ALL = "ALL"
```
Comment (samgo1) [inline uber_sante/models/scheduler.py]: what do you mean by "ALL"?

File: uber_sante/services/availability_service.py [hunk 2]
```diff
@@ -9,63 +9,85 @@ def __init__(self):
...
+
+    def free_availability(self, availability_id):
+
+        update_stmt = 'UPDATE Availability SET free = 1 WHERE id = ?'
+        params = (availability_id, )
+        self.db.write_one(update_stmt, params)
 
     def validate_availability_and_reserve(self, appointment):
+        """
+        Given either a Walkin (20min, 1 row) or Annual appointment (60min, 3 rows),
```
Comment (samgo1) [inline uber_sante/services/availability_service.py]: All appointments are now considered as 1 row with an attribute 'type' for annual or walkin, in the availability table that the doctors will fill with their availabilities

File: uber_sante/models/scheduler.py (hunk 1 above)
Comment (lkadian) [reply uber_sante/models/scheduler.py]: When you make a schedule_request you can request to query for annual, walkin or all

File: uber_sante/services/availability_service.py (hunk 2 above)
Comment (lkadian) [reply uber_sante/services/availability_service.py]: You're right, I wasnt sure about this change since I missed the design. Ill change it

Comment (samgo1) [reply uber_sante/services/availability_service.py]: Some diagrams are updated( the ones with last modified date 16 feb and later), the rest might not reflect these changes yet

**Pull request merged**